### ISSUE TITLE
TimeZone support

### ISSUE THREAD

**Issue by nrinaudo** [47deg/scalacheck-toolbox#16]
I often have use cases where I need to test dates within a specific time zone. It's probably fairly simple to adapt `scalacheck-datetime`'s `Gen` instances to my need - just map into it and set the desired timezone - but then I need to make sure this still falls within the expected date boundaries.

Would it be possible to, say, add a new time zone parameter to `genDateTimeWithinRange`? Might be implicit (probably not a great idea), might have a default value of UTC if you don't want to break backwards compatibility...

**Comment by sullivan-** [comment]
I needed UTC myself, and wrote a little wrapper like so:

```scala
import com.fortysevendeg.scalacheck.datetime.joda.GenJoda.genDateTime                                     
import org.joda.time.DateTimeZone.UTC                                                                     
import org.scalacheck.Arbitrary                                                                           
                                                                                                                   
implicit val arbJodaZ = Arbitrary(genDateTime.map(_.withZone(UTC)))
```

I was thinking of contributing it, probably make it into a def with a `DateTimeZone` parameter with default value of `DateTimeZone.getDefault()`, which would match current behavior. E.g. maybe replace existing

```scala
  implicit val arbJoda: Arbitrary[DateTime] = Arbitrary(genDateTime)
```

with

```scala
  implicit def arbJoda(implicit zone: DateTimeZone = DateTimeZone.getDefault): Arbitrary[DateTime] = Arbitrary(genDateTime.map(_.withZone(zone)))
```

Now users can override the default zone by supplying their own implicit zone.

If you want I can write it up with a test and send a PR. Not sure how much binary compat matters to you; I'm fairly certain this would break it.

**Comment by sullivan-** [comment]
Probably better as

```scala
  implicit def defaultTimeZone = DateTimeZone.getDefault
  implicit def arbJoda(implicit zone: DateTimeZone): Arbitrary[DateTime] =
    Arbitrary(genDateTime.map(_.withZone(zone)))
```

**Comment by sullivan-** [comment]
here's a PR, please take it if you like it. sorry I didn't wait for your feedback before writing it.

@nrinaudo it seems to me you shouldn't have to make changes to `GenDateTime.genDateTimeWithinRange`. this def takes in a `DateTime` as argument, and it looks like the operations in `JodaInstances.jodaForPeriod` will respect the time zone of the original date time passed in.

https://github.com/47deg/scalacheck-toolbox/pull/58

**Comment by juanpedromoreno** [comment]
Fixed by #58
